src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -153,7 +153,7 @@ typedef struct {
   char *rojig_spec;
   char *previous_checksum;
 
-  RORTreefile *treefile_rs;
+  std::optional<rust::Box<rpmostreecxx::Treefile>> treefile_rs;
```

Comment:
Honestly this one took me a while; understanding the "deref semantics" of `std::optional` *and* `rust:Box` and how that interacts with C++ references etc.

Really what makes this code messy is we pass around this context object but we don't always have a "treefile_rs"; it'd be cleaner if we refactored this so there was like a "BaseComposeContext" and a "ComposeContext" but...I also am somewhat hopeful that we can just sprint to rewriting this whole file in Rust.